### PR TITLE
Switch back to the RC2 build since GA has gone internal. 

### PR DESCRIPTION
**Pull request by marcpopMSFT**
Update each month manually.

### CODE REVIEW

**Comment by mmitche** [comment]
That tells me we maybe didn't move the transport packages to the public feed?

File: eng/Version.Details.xml
Comment (mthalman) [inline eng/Version.Details.xml:1]: A whole bunch of internal repo references here.

Comment (mmitche) [reply eng/Version.Details.xml:1]: That would be typical of any release day tag for a monthly release. What's the issue with this? rc2 was built internally.

Comment (mthalman) [reply eng/Version.Details.xml:1]: Ok, wasn't sure. As long as the commits exist in public repos things should be ok.

**Comment by marcpopMSFT** [comment]
@dotnet/source-build-internal I got templating reverted back to rc2 build of runtime and reran this build. The source build leg is failing to find Microsoft.SourceBuild.Intermediate.runtime.linux-x64 which should be the rc2 version of that file. Is it missing from a feed or something?

**Comment by v-wuzhai** [comment]
@dotnet/source-build-internal When I added the option --max-age to 40 days, I got the following query result:
```
PS C:\Users\AzureTest> darc get-asset --name "Microsoft.SourceBuild.Intermediate.runtime.linux-x64" --version 8.0.0-rc.2.23479.6 --max-age 40
Looking up assets with name 'Microsoft.SourceBuild.Intermediate.runtime.linux-x64' and version '8.0.0-rc.2.23479.6' in the last 40 days
Microsoft.SourceBuild.Intermediate.runtime.linux-x64 @ 8.0.0-rc.2.23479.6
Repository:    https://dev.azure.com/dnceng/internal/_git/dotnet-runtime
Branch:        refs/heads/internal/release/frameshift/8.0-rc2
Commit:        0b25e38ad32a69cd83ae246104b32449203cc71c
Build Number:  20230929.6
Date Produced: 9/29/2023 3:44 PM
Build Link:    https://dev.azure.com/dnceng/internal/_build/results?buildId=2280047
AzDO Build Id: 2280047
BAR Build Id:  194966
Released:      False
Channels:
Locations:
- https://dev.azure.com/dnceng/internal/_apis/build/builds/2280047/artifacts (Container)
- https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8-transport/nuget/v3/index.json (NugetFeed)
```

**Comment by mthalman** [comment]
@mmitche - Do you know why Microsoft.SourceBuild.Intermediate.runtime.linux-x64.8.0.0-rc.2.23479.6 doesn't exist in the transport feed and how to get it there?

**Comment by MichaelSimons** [comment]
> @dotnet/source-build-internal When I added the option --max-age to 40 days, I got the following query result:
> 
> ```
> PS C:\Users\AzureTest> darc get-asset --name "Microsoft.SourceBuild.Intermediate.runtime.linux-x64" --version 8.0.0-rc.2.23479.6 --max-age 40
> Looking up assets with name 'Microsoft.SourceBuild.Intermediate.runtime.linux-x64' and version '8.0.0-rc.2.23479.6' in the last 40 days
> Microsoft.SourceBuild.Intermediate.runtime.linux-x64 @ 8.0.0-rc.2.23479.6
> Repository:    https://dev.azure.com/dnceng/internal/_git/dotnet-runtime
> Branch:        refs/heads/internal/release/frameshift/8.0-rc2
> Commit:        0b25e38ad32a69cd83ae246104b32449203cc71c
> Build Number:  20230929.6
> Date Produced: 9/29/2023 3:44 PM
> Build Link:    https://dev.azure.com/dnceng/internal/_build/results?buildId=2280047
> AzDO Build Id: 2280047
> BAR Build Id:  194966
> Released:      False
> Channels:
> Locations:
> - https://dev.azure.com/dnceng/internal/_apis/build/builds/2280047/artifacts (Container)
> - https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet8-transport/nuget/v3/index.json (NugetFeed)
> ```

Looks like the dotnet8-transport feed is missing from the nuget.config.

**Comment by mthalman** [comment]
> Looks like the dotnet8-transport feed is missing from the nuget.config.

It's looking there but not finding that version.

```
  /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/8.0.0-beta.23525.4/tools/Tools.proj : error NU1102:   - Found 1272 version(s) in dotnet8-transport [ Nearest version: 8.0.0-rtm.23472.12 ]
```

**Comment by MichaelSimons** [comment]
> > Looks like the dotnet8-transport feed is missing from the nuget.config.
> 
> It's looking there but not finding that version.
> 
> ```
>   /__w/1/s/.packages/microsoft.dotnet.arcade.sdk/8.0.0-beta.23525.4/tools/Tools.proj : error NU1102:   - Found 1272 version(s) in dotnet8-transport [ Nearest version: 8.0.0-rtm.23472.12 ]
> ```

I saw it immediately after I said that :)

**Comment by mmitche** [comment]
Maybe the intenral transport packages didn't get pushed on release day? @tkapin @rbhanda. I think this worked for rc1 though.

**Comment by tkapin** [comment]
> Maybe the intenral transport packages didn't get pushed on release day? @tkapin @rbhanda. I think this worked for rc1 though.

@rahul, can you investigate this or you need my folks to look into this? If so, please let me know so they can look into this in the morning.

**Pull request closed without merging**